### PR TITLE
Trendy: delta2kernel and delta2lr

### PR DESCRIPTION
**Pull request by StevenM1**
- Added the double-learning rate delta rule (with a positive and negative learning rate) as 'delta2lr';
- Renamed 'delta2' to 'delta2kernel' to avoid confusion;
- Minor change to the help function ('Trend options' -> 'Phase options'; for consistency with new name)

**Considerations for the future**
The two–learning-rate delta rule uses separate learning rates for positive and negative prediction errors. For example:

```
    pe[i-1] = covariate[i-1] - q[i-1];
    learning_rate = (pe[i-1] > 0) ? alphaPos[i-1] : alphaNeg[i-1];
    q[i] = q[i-1] + learning_rate * pe[i-1];
```

At present, this is implemented as a separate rule with two parameters: `alphaPos` and `alphaNeg`.

#### **Motivation**

Ideally, we’d like to use a mean–difference parametrization to alpha instead of two independent 'absolute' parameters. That is, define:

* a **mean learning rate** (`alphaMean`) (estimated on the probit scale)
* a **difference parameter** (`alphaDifference` = positive – negative) (estimated on the natural scale)

This would be advantageous because:

* The difference is often a parameter of theoretical interest.
* It allows for more flexible modeling (e.g., setting priors, varying by condition, or using a Savage–Dickey ratio test).

**Challenges**

1. **Parallel with contrasts or design matrices**
   Conceptually, this is similar to how contrasts are mapped (e.g., with an `ADmat` for a factor like `PEsign`).
   However, `PEsign` (the sign of the prediction error) isn’t known until the kernel is run.
   That means we can’t simply include `PEsign` as a factor in a hypothetical formula for `alpha`—it’s a runtime quantity.

2. **Transformations and parameter dependencies**
   Alternatively, we could define `alphaMean` and `alphaDifference` directly as kernel parameters and then derive (when running the rule):

   ```
   alphaPos = pnorm(alphaMean + alphaDifference)
   alphaNeg = pnorm(alphaMean - alphaDifference)
   ```
   The issue is that transformations are applied **before** the kernel is evaluated, so they would act on `alphaMean` and `alphaDifference` individually rather than their combinations (e.g., `alphaPos = pnorm(alphaMean) + alphaDifference`).

So not sure how to proceed there.. Any suggestions?

### CODE REVIEW

**Comment by StevenM1** [comment]
With a bit of creativity, a mean-difference parametrisation is actually possible with creative use of trends:

```
trendMeans <- make_trend(par_names=c('v_lMd.alphaPos', 'v_lMd.alphaNeg'),
                         cov_names='mean_',
                         kernels=c('lin_incr', 'lin_incr'),
                         bases=c('lin', 'lin'))
trendDiffs <- make_trend(par_names=c('v_lMd.alphaPos', 'v_lMd.alphaNeg'),
                         cov_names='diff_',
                         kernels=c('lin_incr', 'lin_decr'),
                         bases=c('lin', 'lin'))
trend2lr <- make_trend(par_names = "v_lMd", cov_names = c("rewards"), kernels = "delta2lr", bases='lin', at='lR')
trend <- list(v_lMd.alphaPos=trendMeans$v_lMd.alphaPos,
              v_lMd.alphaPos=trendDiffs$v_lMd.alphaPos,
              v_lMd.alphaNeg=trendMeans$v_lMd.alphaNeg,
              v_lMd.alphaNeg=trendDiffs$v_lMd.alphaNeg,
              v_lMd=trend2lr$v_lMd)
trend[[1]]$trend_pnames <- 'v_lMd.alphaMean'
trend[[2]]$trend_pnames <- 'v_lMd.alphaDiff'
trend[[3]]$trend_pnames <- 'v_lMd.alphaMean'
trend[[4]]$trend_pnames <- 'v_lMd.alphaDiff'
attr(trend, 'sequential') <- TRUE

design_base <- design(factors = list(subjects = 1, S = 1:2),
                      Rlevels = 1:2,
                      covariates = c('rewards', 'mean_', 'diff_'),
                      matchfun = matchfun,
                      trend = trend,
                      formula = list(v ~ lM, s ~ 1, t0 ~ 1,
                                     v_lMd.alphaPos~1, v_lMd.alphaNeg~1, v_lMd.alphaMean~1, v_lMd.alphaDiff~1),
                      contrasts = list(lM = ADmat),
                      constants = c('v_lMd.q0'=0, s=log(1), v_lMd.alphaPos=0, v_lMd.alphaNeg=0, v_lMd=0),
                      model = RDM)

## double check - does this work?
p_vector <- sampled_pars(design_base,  doMap = FALSE)
p_vector[1:5] <- c(-1, log(.2), 
qnorm(0.2), # alpha mean centered at 0.2
 1, # alpha difference (pos - neg) is 1 on the probit scale
1) 

n_trials <- 1000
rewards <- rbinom(n_trials, 1, .7)
dat <- make_data(p_vector, design_base, n_trials = n_trials, covariates = data.frame(rewards=rbinom(n_trials, 1, .7),
                                                                                     mean_=1,
                                                                                     diff_=0.5))

emc <- make_emc(dat, design_base, type='single', compress=FALSE)
par(mfrow=c(2,1))
plot_trend(p_vector, emc, par_name='v', filter=function(x) x$lM==TRUE, xlim=c(1, 50))
plot_trend(p_vector, emc, par_name='v', filter=function(x) x$lM==FALSE, xlim=c(1, 50))
```
